### PR TITLE
drivers: net: ppp: async uart support

### PR DESCRIPTION
**Pull request by jhirsi**
Implemented an option to use asynchronous uart api in ppp driver (CONFIG_NET_PPP_ASYNC_UART).

Signed-off-by: Jani Hirsimäki <jani.hirsimaki@nordicsemi.no>

### CODE REVIEW

File: drivers/net/ppp.c [hunk 1]
```diff
@@ -206,6 +365,18 @@ static int ppp_send_flush(struct ppp_driver_context *ppp, int off)
 	 */
 	if (IS_ENABLED(CONFIG_GSM_MUX)) {
 		(void)uart_fifo_fill(ppp->dev, buf, off);
+	} else if (IS_ENABLED(CONFIG_NET_PPP_ASYNC_UART)) {
```
Comment (jukkar) [inline drivers/net/ppp.c:368]: This seems to be incompatible with muxing, could you add suitable lines to Kconfig so that this is enforced (user cannot select both at the same time)

Comment (jhirsi) [reply drivers/net/ppp.c:368]: Good point. Added.

**Review by jukkar — APPROVED** [review]
LGTM

File: drivers/net/ppp.c [hunk 2]
```diff
@@ -30,13 +30,15 @@ LOG_MODULE_REGISTER(net_ppp, LOG_LEVEL);
 #include <sys/ring_buffer.h>
 #include <sys/crc.h>
 #include <drivers/uart.h>
+
```
Comment (rlubos) [inline drivers/net/ppp.c:33]: nit: Was this addition intended?

**Review by rlubos — APPROVED** [review]
LGTM, just one nit.

File: drivers/net/ppp.c (hunk 2 above)
Comment (jhirsi) [reply drivers/net/ppp.c]: Removed added extra line.
